### PR TITLE
"[E] to enable building" hint while building is paused

### PR DESCRIPTION
**Pull request by joshuaptfan**
Currently, when "auto-pause building" is off, no UI hint is shown when players press E to toggle pause/resume building, until they try to build something. Thus, they are left to guess what the current state of building is if they have pressed E multiple times. If they guess wrong, this can cause them to accidentally start building something when they only wanted to create a plan.

This PR makes it so that an "[E] to enable building" hint is a persistent indicator of the state of building (always displayed when building is paused) when auto-pause is off, regardless if building plans exist. It does not change how the hint is displayed when auto-pause is on.

### CODE REVIEW

**Comment by joshuaptfan** [comment]
Ideally, "[Q] to clear plan, [F] to select+copy" would only display when building plans exist, making the building pause indicator less obtrusive. I'm currently having trouble implementing this.

**Comment by joshuaptfan** [comment]
New smaller hint
enable building

File: core/src/mindustry/input/DesktopInput.java [hunk 1]
```diff
@@ -68,14 +68,28 @@ public void buildUI(Group group){
...
                 b.row();
                 b.label(() -> Core.bundle.format("cancelbuilding", Core.keybinds.get(Binding.clear_building).key.toString())).style(Styles.outlineLabel);
                 b.row();
                 b.label(() -> Core.bundle.format("selectschematic", Core.keybinds.get(Binding.schematic_select).key.toString())).style(Styles.outlineLabel);
             }).margin(10f);
         });
 
+        //build enable hint
+        group.fill(t -> {
+            t.bottom();
```
Comment (Anuken) [inline core/src/mindustry/input/DesktopInput.java:81]: You should probably initialize this table with `t.color.a = 0` so it starts out hidden.

Comment (joshuaptfan) [reply core/src/mindustry/input/DesktopInput.java:81]: Then shouldn't that also be applied to the general building hint table above?

Comment (Anuken) [reply core/src/mindustry/input/DesktopInput.java:82]: The difference is that this one starts out visible, while the other hints dialog does not.

Comment (joshuaptfan) [reply core/src/mindustry/input/DesktopInput.java:81]: Testing without `t.color.a = 0`, that line doesn't seem to be necessary. Tested these scenarios:

* Starting fresh custom game with auto-pause off: build enable hint not visible until pressing E.
* Resuming saved game with building disabled: build enable hint visible as expected, disappears when pressing E.
* Resuming saved game with building enabled: build enable hint not visible as expected, appears when pressing E.
* Changing auto-pause setting during game: some anomalous behavior, like having to press E twice to get hint to appear.

[`isBuilding` starts out `true`](https://github.com/Anuken/Mindustry/blob/0cc9b0b0a3b7a30ae1dab30d3f194ac04f7152fc/core/src/mindustry/input/InputHandler.java#L60), which means the build enable hint is invisible at the start of the game.

Comment (joshuaptfan) [reply core/src/mindustry/input/DesktopInput.java:81]: Fixed inconsistencies by replacing `!buildWasAutoPaused` with `!Core.settings.getBool("buildautopause")`. Now the build enable hint always reflects that state of `isBuilding` accurately, when auto-pause is disabled.

**Comment by joshuaptfan** [comment]
All edge cases have been fixed as far as I can tell, with minimal increase in code. The PR currently passes all these tests:
* Press E to toggle building multiple times
  * "[E] to enable building" should fade out
* Build something
  * "[E] to pause building" should fade out
* Pause game, create build plan, toggle building multiple times, unpause, build
  * "[E] to pause building" should fade out
* Enable auto-pause and build
  * "[E] to pause/resume building" should appear as expected
  * "[E] to enable building" should never appear
* Disable auto-pause
  * "[E] to enable building" should appear immediately, and disappear when E is pressed

This partial state machine may help with understanding some of the logic. It shows the necessity of adding the `wasBuilding` variable to track the previous state of `isBuilding`.

> build disabled
> player not building
> "[E] to enable building"
> 
> build disabled -> build enabled
> player not building
> "[E] to enable building" fade out
> 
> build enabled
> player building
> "[E] to pause/resume building"
> 
> build enabled
> player building -> player not building
> "[E] to pause/resume building" fade out
> 
> build enabled
> player not building
> (hidden)

**Comment by Anuken** [comment]
As reported in #4642, this draws over the respawn text. I don't see an easy fix for this, so I'm reverting this change until further notice.
